tests/astimezone_test.py

```diff
@@ -0,0 +1,247 @@
+    assert before_dst_cet.hour == 3  # 1:30 UTC + 1 hour = 3:30 CET
+    assert after_dst_cet.hour == 4   # 2:30 UTC + 2 hours = 4:30 CEST
+
+
+def test_astimezone_same_timezone() -> None:
+    """Test astimezone when converting to the same timezone."""
+    dt = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+
+    # Convert to the same timezone
+    result = dt.astimezone(ZoneInfo('UTC'))
+
+    assert isinstance(result, datetime_tz)
+    assert result == dt  # Should be identical
+    assert result.tzinfo == ZoneInfo('UTC')
+
+
+def test_astimezone_with_different_datetime_types() -> None:
+    """Test astimezone behavior with different datetime_tz subclasses."""
+    # Test datetime_utc
+    utc_dt = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+    utc_to_cet = utc_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(utc_to_cet, datetime_tz)
+    assert utc_to_cet.tzinfo == ZoneInfo('CET')
+
+    # Test datetime_cet
+    cet_dt = datetime_cet(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET'))
+    cet_to_utc = cet_dt.astimezone(ZoneInfo('UTC'))
+    assert isinstance(cet_to_utc, datetime_tz)
+    assert cet_to_utc.tzinfo == ZoneInfo('UTC')
+
+    # Test datetime_local
+    local_dt = datetime_local.now()
+    local_to_cet = local_dt.astimezone(ZoneInfo('CET'))
+    assert isinstance(local_to_cet, datetime_tz)
+    assert local_to_cet.tzinfo == ZoneInfo('CET')
+
+
+def test_astimezone_timezone_conversion_accuracy() -> None:
+    """Test that astimezone correctly handles timezone conversions."""
+    # Test UTC to various timezones
+    utc_dt = datetime_utc(2021, 1, 1, 12, 0, 0, tzinfo=ZoneInfo('UTC'))
+
+    # UTC to EST (UTC-5)
+    est_result = utc_dt.astimezone(ZoneInfo('EST'))
+    assert est_result.hour == 7  # 12:00 UTC - 5 hours = 7:00 EST
+
+    # UTC to CET (UTC+1 in January)
+    cet_result = utc_dt.astimezone(ZoneInfo('CET'))
+    assert cet_result.hour == 13  # 12:00 UTC + 1 hour = 13:00 CET
+
+    # UTC to JST (UTC+9)
+    jst_result = utc_dt.astimezone(ZoneInfo('Asia/Tokyo'))
+    assert jst_result.hour == 21  # 12:00 UTC + 9 hours = 21:00 JST
+
+
+def test_astimezone_without_arguments() -> None:
+    """Test that astimezone without arguments defaults to local timezone."""
+    # Test with datetime_utc
+    utc_dt = datetime_utc(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('UTC'))
+    result = utc_dt.astimezone()  # No arguments
+
+    assert isinstance(result, datetime_tz)
+    # Should convert to local timezone
+    assert result.tzinfo is not None
+    assert result.tzinfo != ZoneInfo('UTC')  # Should not be UTC anymore
+
+    # Test with datetime_cet
+    cet_dt = datetime_cet(2021, 1, 1, 10, 0, 0, tzinfo=ZoneInfo('CET'))
+    result_cet = cet_dt.astimezone()  # No arguments
+
+    assert isinstance(result_cet, datetime_tz)
+    # Should convert to local timezone
+    assert result_cet.tzinfo is not None
+    assert result_cet.tzinfo != ZoneInfo('CET')  # Should not be CET anymore
```

Comment:
This test only works as long as the machine running the test is not in the `CET` timezone 🙈